tests/components/application-connector/test/compass-runtime-agent/secret.go

```diff
@@ -0,0 +1,107 @@
+package compass_runtime_agent
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+	"reflect"
+	"testing"
+)
+
+var secretRepos map[string]v1.SecretInterface
+
+type secretData struct {
+	password     string
+	username     string
+	crt          string
+	key          string
+	clientId     string
+	clientSecret string
+}
+
```

Comment:
You are missing unit tests for the SecretClient interface and Compare function. 

To implement it you can use fake client to make test secrets . "k8s.io/client-go/kubernetes/fake"

Example:

```
	import "k8s.io/client-go/kubernetes/fake"


         coreV1 := fake.NewSimpleClientset()
	secrets := coreV1.CoreV1().Secrets(namespace)

         createFakeCredentialsSecret(t, secrets, credentials)


func createFakeCredentialsSecret(t *testing.T, secrets core.SecretInterface, credentials credentials) {

	secret := &v1.Secret{
		ObjectMeta: meta.ObjectMeta{
			Name:      secretName,
			Namespace: namespace,
		},
		TypeMeta: meta.TypeMeta{
			Kind:       "Secret",
			APIVersion: "v1",
		},
		Data: map[string][]byte{
			"key1":       []byte("val1"),
			"key2":   []byte("val2"),
			"key3": []byte("val3"),
		},
	}

	_, err := secrets.Create(context.Background(), secret, meta.CreateOptions{})

	require.NoError(t, err)
}

```